src/thrift-client/TgtInterfaceImpl.cpp

```diff
@@ -1068,7 +1100,89 @@ void StordVmdk::ScheduleBulkRead(folly::EventBase* basep,
 		BulkReadComplete(requests, result);
 		--stats_.rpc_requests_scheduled_;
 	});
+}
+
+folly::Future<int> StordVmdk::ScheduleTruncate(StorRpcAsyncClient* clientp,
+		RequestID reqid, std::vector<TruncateReq>&& requests) {
 	++stats_.rpc_requests_scheduled_;
+	return clientp->future_Truncate(vmdk_handle_, reqid, std::forward<std::vector<TruncateReq>>(requests))
+	.then([this] (const TruncateResult& result) {
+		--stats_.rpc_requests_scheduled_;
+		return result.get_result();
+	})
+	.onError([this] (const std::exception& e) {
+		--stats_.rpc_requests_scheduled_;
+		return -EIO;
+	});
+}
+
+void StordVmdk::ScheduleTruncate(folly::EventBase* basep,
+		StorRpcAsyncClient* clientp, Request* reqp) {
+	log_assert(basep->isInEventBaseThread());
+
+	std::vector<folly::Future<int>> futures;
+	std::vector<::hyc_thrift::TruncateReq> truncate;
+
+	auto length = reqp->buf_sz;
+	char* bufp = reqp->bufferp;
+	for (; length >= 16; length -= 16, bufp += 16) {
+		uint64_t offset;
+		uint32_t len;
+
+		offset = BigEndian::DeserializeInt<decltype(offset)>
```

Comment:
comment from @dmehta85. If running architecture is BE then we do not need serialization.

Reply:
@dmehta85 this code is correct. Deserilization is needed to convert number in series of uint8_t* to uint64_t.